resolver/src/main/java/io/netty/resolver/RoundRobinInetAddressResolver.java

```diff
@@ -64,7 +64,7 @@ public void operationComplete(Future<List<InetAddress>> future) throws Exception
                     if (numAddresses > 0) {
                         // if there are multiple addresses: we shall pick one by one
                         // to support the round robin distribution
-                        promise.setSuccess(inetAddresses.get(index.getAndIncrement() % numAddresses));
+                        promise.setSuccess(inetAddresses.get(toIndex(counter.getAndIncrement(), numAddresses)));
```

Comment:
@Spikhalskiy @guidomedina maybe I am missing something but why can't we just use `Math.abs(index.getAndIncrement() % numAddresses)` ?